test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpHandler.java

```diff
@@ -145,83 +148,110 @@ public void handle(final HttpExchange exchange) throws IOException {
                 exchange.getResponseHeaders().add("x-ms-request-server-encrypted", "false");
                 exchange.sendResponseHeaders(RestStatus.CREATED.getStatus(), -1);
 
+            } else if (Regex.simpleMatch("PUT /" + account + "/" + container + "*comp=lease*", request)) {
+                // Lease Blob (https://learn.microsoft.com/en-us/rest/api/storageservices/lease-blob)
+                final String leaseAction = requireHeader(exchange, "x-ms-lease-action");
+
+                switch (leaseAction) {
+                    case "acquire" -> {
+                        final int leaseDurationSeconds = requireIntegerHeader(exchange, X_MS_LEASE_DURATION);
+                        final String proposedLeaseId = exchange.getRequestHeaders().getFirst(X_MS_PROPOSED_LEASE_ID);
+                        final String newLeaseId = mockAzureBlobStore.acquireLease(
+                            blobPath(exchange),
+                            leaseDurationSeconds,
+                            proposedLeaseId
+                        );
+                        exchange.getResponseHeaders().set(X_MS_LEASE_ID, newLeaseId);
+                        exchange.sendResponseHeaders(RestStatus.CREATED.getStatus(), -1);
+                    }
+                    case "release" -> {
+                        final String leaseId = requireHeader(exchange, X_MS_LEASE_ID);
+                        mockAzureBlobStore.releaseLease(blobPath(exchange), leaseId);
+                        exchange.sendResponseHeaders(RestStatus.OK.getStatus(), -1);
+                    }
+                    case "renew", "change", "break" -> throw new MockAzureBlobStore.AzureBlobStoreError(
+                        RestStatus.NOT_IMPLEMENTED,
+                        "NotImplemented",
+                        "Attempted to use unsupported lease API: " + leaseAction
+                    );
+                    default -> throw new MockAzureBlobStore.BadRequestException(
+                        "InvalidHeaderValue",
+                        "Invalid x-ms-lease-action header: " + leaseAction
+                    );
+                }
             } else if (Regex.simpleMatch("PUT /" + account + "/" + container + "/*", request)) {
                 // PUT Blob (see https://docs.microsoft.com/en-us/rest/api/storageservices/put-blob)
-                final String ifNoneMatch = exchange.getRequestHeaders().getFirst("If-None-Match");
-                if ("*".equals(ifNoneMatch)) {
-                    if (blobs.putIfAbsent(exchange.getRequestURI().getPath(), Streams.readFully(exchange.getRequestBody())) != null) {
-                        sendError(exchange, RestStatus.CONFLICT);
-                        return;
-                    }
-                } else {
-                    blobs.put(exchange.getRequestURI().getPath(), Streams.readFully(exchange.getRequestBody()));
+                final String blobTypeHeader = requireHeader(exchange, X_MS_BLOB_TYPE);
```

Comment:
For my understanding: Does this change mean that we used to assume every PUT request is for Block blobs and now we are tightening it up by actually checking the specified blob type which should be configured by the client?

Reply:
Yeah that parameter is required, so we should only ever be creating block blobs.

Reply:
https://learn.microsoft.com/en-us/rest/api/storageservices/understanding-block-blobs--append-blobs--and-page-blobs